go/service/account.go

```diff
@@ -276,14 +277,49 @@ func (h *AccountHandler) TimeTravelReset(ctx context.Context, arg keybase1.TimeT
 	return err
 }
 
-func (h *AccountHandler) UserGetContactSettings(ctx context.Context) (ret keybase1.ContactSettings, err error) {
+func (h *AccountHandler) UserGetContactSettings(ctx context.Context) (res keybase1.ContactSettings, err error) {
 	mctx := libkb.NewMetaContext(ctx, h.G())
 	defer mctx.TraceTimed("AccountHandler#UserGetContactSettings", func() error { return err })()
-	return libkb.GetContactSettings(mctx)
+	res, err = libkb.GetContactSettings(mctx)
+	for i, team := range res.Teams {
+		name, err := teams.ResolveIDToName(mctx.Ctx(), mctx.G(), *team.TeamID)
+		if err != nil {
+			return res, fmt.Errorf("Unexpected error resolving team ID (%v) to name", team.TeamID)
+		}
+		nameStr := name.String()
+		res.Teams[i].TeamName = &nameStr
+	}
+	return res, err
 }
 
+// UserSetContactSettings accepts TeamContactSettings with either TeamID or
```

Comment:
this change was made to support CLI use (so that one can input the team name rather than the team id). i did not want to put this resolve logic in the CLI because the CLI interface is just ferrying json blobs; it shouldn't need to know any details about what's in keybase1.ContactSettings.

Reply:
Resolve logic is ok to be in the CLI